feat: Add OpenID Connect integration to 12 Factor (#2373)
canonical/charmcraft: docs/reuse/reference/extensions/integrations.rst

              optional: True
              limit: 1

  * - `Hydra <https://charmhub.io/hydra>`__ charm

>     - The ``oauth`` interface is used to provide the OpenID Connect authentication
>       protocol. The endpoint name will be used as the prefix to the configuration
>       options related to this endpoint.

Reviewer: ```suggestion
    - The ``oauth`` interface is a conduit for the OpenID Connect authentication
      protocol. Each configuration option for this endpoint is prefixed with the
      endpoint name.
```

      .. code-block:: yaml

          requires:
            oauth-endpoint-name:
              interface: oauth
              optional: True
              limit: 1

>       The integration will create two configuration options,

Reviewer: > integration

Should this be "relation"?
